src/lint/concept_exercises.nim

```diff
@@ -1,29 +1,40 @@
-import std/[json, os, terminal]
+import std/[json, os]
 import ".."/helpers
 import "."/validators
 
 proc isValidAuthorOrContributor(data: JsonNode, context: string, path: string): bool =
   if isObject(data, context, path):
     result = true
-    checkString("github_username")
-    checkString("exercism_username", isRequired = false)
+    if not checkString(data, "github_username", path):
+      result = false
+    if not checkString(data, "exercism_username", path, isRequired = false):
+      result = false
 
-template checkFiles(data: JsonNode, context, path: string) =
+proc checkFiles(data: JsonNode, context, path: string): bool =
+  result = true
   if hasObject(data, context, path):
-    checkArrayOfStrings(context, "solution")
-    checkArrayOfStrings(context, "test")
-    checkArrayOfStrings(context, "exemplar")
+    if not checkArrayOfStrings(data, context, "solution", path):
+      result = false
+    if not checkArrayOfStrings(data, context, "test", path):
+      result = false
+    if not checkArrayOfStrings(data, context, "exemplar", path):
+      result = false
   else:
     result = false
 
 proc isValidConceptExerciseConfig(data: JsonNode, path: string): bool =
   if isObject(data, "root", path):
     result = true
-    checkArrayOf("authors", isValidAuthorOrContributor)
-    checkArrayOf("contributors", isValidAuthorOrContributor, isRequired = false)
-    checkFiles(data, "files", path)
-    checkArrayOfStrings("", "forked_from", isRequired = false)
-    checkString("language_versions", isRequired = false)
+    if not checkArrayOf(data, "authors", path, isValidAuthorOrContributor):
+      result = false
+    if not checkArrayOf(data, "contributors", path, isValidAuthorOrContributor, isRequired = false):
+      result = false
+    if not checkFiles(data, "files", path):
+      result = false
+    if not checkArrayOfStrings(data, "", "forked_from", path, isRequired = false):
```

Comment:
I still don't like the `""` argument being required. I'd like to refactor in the future to not having to require that. But it's fine for now.